releasenotes/notes/0.15/quadratic-form-circuit-f0ddb1694960cbc1.yaml

```
@@ -0,0 +1,22 @@
+---
+features:
+  - |
+    A new class, :class:`~qiskit.circuit.library.QuadraticForm` to the
+    :mod:`qiskit.circuit.library` module for implementing a a quadratic form on
+    binary variables. The circuit library element implements the operation
+
+    .. math::
+
+      |x\rangle |0\rangle \mapsto |x\rangle |Q(x) \mod 2^m\rangle
+
+    for the quadratic form :math:`Q` and :math:`m` output qubits.
+    The quadratic form is specified using a matrix for the quadratic
+    terms, a vector for the linear terms and a constant offset. For example::
+
+      import numpy as np
+
+      from qiskit.circuit.library import QuadraticForm
+
+      A = np.array([[1.0, 0.0], [0.0, 1.0]])
+      b = np.array([.5, -.5])
+      quad_form_circuit = QuadraticForm(2, A, b)
```

This example might be better because it contains quadratic terms (diagonals of A are linear terms because the variables are binary and therefore `x_i * x_i = x_i`) and it can be implemented exactly because all terms are integers.
```suggestion
      A = np.array([[1, 2], [-1, 0]])
      b = np.array([3, -3])
      c = -2
      m = 4
      quad_form_circuit = QuadraticForm(m, A, b, c)
```